### ISSUE TITLE
can't mask existing function name with inner function of same name.

### ISSUE THREAD

**Issue by Dunbaratu** [KSP-KOS/KOS#777]
Example case showing the problem:
```
// An example testing nesting a function inside a function
// when both of them end up having the same name:
// Proper nesting rules should mask the outer function
// with the inner function.

declare function samename {
  declare x to 1.
  print "outer function samename() has x = " + x.

  declare function samename {
    declare x to 2.
    print "inner function samename() has x = " + x.
  }.
  samename().  // should call the inner samename() here.
}.

samename(). // should call the outer samename() here.
```

Given proper scoping rules, the code SHOULD result in the output:
```
outer function samename() has x = 1
inner function samename() has x = 2
```
But instead it calls directly into the inner function only from outside the outer function, giving output of just this one line:
```
inner function samename() has x = 2
```

The problem happens because the compiler uses a key value of JUST the function's name alone as the hash key for referring to the function body, with no regard to nesting or scoping.  Thus it cannot build two different function bodies both called 'foo'.  The second instance will clobber over the top of the first instead of making a new one.

**Comment by Dunbaratu** [comment]
okay, I THINK I solved this with a bit of rework, but as it's in the middle of my scope changes branch I can't test until that gets done.

Reminder to self: re-test after merge of #750 and see if it's solved.

file: functest14.ks contains the above test, and will be included in the #750 PR.

**Comment by Dunbaratu** [comment]
fixed by pr #750
